Address section 9 edits and add common definition for reason phrase (#956)
moq-wg/moq-transport: draft-ietf-moq-transport.md

the receiver MUST terminate the session with error code 'Key-Value
Formatting Error'.

### Reason Phrase Structure {#reason-phrase}

Reason Phrase provides a way for the sender to encode additional diagnostic
information about the error condition, where appropriate.

~~~
Reason Phrase {
  Reason Phrase Length (i),
  Reason Phrase Value (..)
}
~~~

* Reason Phrase Length: A variable-length integer specifying the length of the
  reason phrase in bytes. The reason phrase length has a maximum length of

>   1024 bytes. If an endpoint receives a length exceeding the maximum, it MUST

Reviewer: If the length is max 1024 bytes, can we use a 2 byte length field instead of a variable length integer? (I see we had this restriction before, and changing the field here would make this non-editorial. Maybe we do it in a different PR?)

Reviewer 2: I think we discussed this before and decided it'd be best to do it in a separate PR.

> * Reason Phrase Length: A variable-length integer specifying the length of the
> reason phrase in bytes. The  reason phrase length has a maximum length of
> 1024 bytes. If an endpoint recieves a length exceeding the maximum, it MUST
> close the session with a Protocol Violation
> 
> * Reason Phrase Value: Additional diagnostic information about the error condition.
> The reason phrase value is encoded as UTF-8 string and does not carry information,
> such as language tags, that would aid comprehension by  any entity other than
> the one that created the text.

Reviewer 3: ```suggestion
* Reason Phrase Length: A variable-length integer specifying the length of the
  reason phrase in bytes. The reason phrase length has a maximum length of
  1024 bytes. If an endpoint receives a length exceeding the maximum, it MUST
  close the session with a Protocol Violation

* Reason Phrase Value: Additional diagnostic information about the error condition.
  The reason phrase value is encoded as UTF-8 string and does not carry information,
  such as language tags, that would aid comprehension by any entity other than
  the one that created the text.
```

Fields present only for Relative Fetch (0x2) and Absolute Fetch (0x3):

> * Joining Subscribe ID: The Subscribe ID of the existing subscription to be
>   joined. If a publisher receives a Joining Fetch with a Subscribe ID that does
>   not correspond to an existing Subscribe in the same session, it MUST respond
>   with a Fetch Error with code Invalid Subscribe ID.

Reviewer 3: Subscribe ID is no longer a term in the document, but we can call this Joining Subscribe ID -- if you are changing the error code name please also update the section in FETCH_ERROR.

```suggestion
* Joining Subscribe ID: The Request ID of the existing subscription to be
  joined. If a publisher receives a Joining Fetch with a Request ID that does
  not correspond to an existing Subscribe in the same session, it MUST respond
  with a Fetch Error with code Invalid Joining Subscribe ID.
```

  Length (i),
  Track Namespace (tuple),
  Error Code (i),

>   Erorr Reason (Reason Phrase),

Reviewer 2: ```suggestion
  Error Reason (Reason Phrase),
```

violation. Objects can arrive after a subscription or fetch has been cancelled,
so the session MUST NOT be teriminated in that case.

Every Track has a single 'Object Forwarding Preference'. If a subscriber
receives objects via both stream and datagram in response to a SUBSCRIBE, or
via Datagram in response to a FETCH, it SHOULD close the session with an error

> of 'Protocol Violation."

Reviewer 3: I forgot we had this text.  Changing the normative language here makes this PR no longer editorial, let's make the change in #938 instead

> Every Track has a single 'Object Forwarding Preference'. If a subscriber
> receives objects via both stream and datagram in response to a SUBSCRIBE, or
> via Datagram in response to a FETCH, it SHOULD close the session with an error
> of 'Protocol Violation."

Reviewer 3: ```suggestion
Every Track has a single 'Object Forwarding Preference' and the Original
Publisher MUST NOT mix different forwarding preferences within a single track.
If a subscriber receives Objects via both Subgroup streams and Datagrams in
response to a SUBSCRIBE, it SHOULD close the session with an error of 'Protocol Violation."
```

Every Track has a single 'Object Forwarding Preference' and the Original
Publisher MUST NOT mix different forwarding preferences within a single track.
If a subscriber receives Objects via both Subgroup streams and Datagrams in

> response to a SUBSCRIBE, it SHOULD close the session with an error of 'Protocol Violation."

Reviewer 3: ```suggestion
response to a SUBSCRIBE, it SHOULD close the session with an error of 'Protocol
Violation'
```

the Object {{priorities}}.

* Object Forwarding Preference: An enumeration indicating how a publisher sends

> an object. The preferences are Subgroup and Datagram.  When publishing, an Object

Reviewer 3: When is an object sent other than when it is published? Fetch response?

Author: yeah . this was just to clarify the context. I am fine either ways.

> an object. The preferences are Subgroup and Datagram.  When publishing, an Object
> MUST be sent according to its `Object Forwarding Preference`, described below.

Reviewer 3: ```suggestion
an object. The preferences are Subgroup and Datagram.  When in response to a
SUBSCRIBE, an Object MUST be sent according to its `Object Forwarding
Preference`, described below.
```
